Warden/Utils/StringHelpers.cs

```diff
@@ -0,0 +1,235 @@
+        /// <remarks>
+        ///  <see cref="Uri.IsWellFormedUriString"/> will fail on uris such as this "spotify://album:27ftYHLeunzcSzb33Wk1hf" even though Windows is capable of launching it still.
+        ///  To work around that we use regex to extra the scheme as a fallback.
+        /// </remarks>
+        internal static string GetUriScheme(this string uri)
+        {
+            if (Uri.IsWellFormedUriString(uri, UriKind.Absolute))
+            {
+                return new Uri(uri).Scheme;
+            }
+            if (Regex.Match(uri, @"^[^:]+(?=:\/\/)", RegexOptions.Compiled | RegexOptions.IgnoreCase) is {Success: true} match)
+            {
+                return match.Value.Trim();
+            }
+            return string.Empty;
+        }
+
+        /// <summary>
+        ///     Attempts to normalize a given directory or file path to a consistent standard by removing potential escaped characters, inconsistent directory delimiters, and more.
+        /// </summary>
+        internal static string NormalizePath(this string path)
+        {
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                return string.Empty;
+            }
+            // if the input is a valid URI return so we don't mess it up.
+            if (IsValidUri(path) && !IsUncPath(path))
+            {
+                return !path.EndsWith("//") ? $"{path}//" : path;
+                //throw new WardenException($"\"{path}\" does not meet the requirements for a Warden path. Only valid application URIs or the fully qualified name of a file or directory are allowed.");
+            }
+            // work around an issue where the below code doesn't like directories that are just "C:"
+            if (path.Length == 2 && path.EndsWith(":"))
+            {
+                path = $"{path}\\";
+            }
+            return Path.GetFullPath(new Uri(path).LocalPath).TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
+        }
+
+
+        /// <summary>
+        /// Parses and properly formats a command-line string.
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="processImage"></param>
+        /// <returns></returns>
+        internal static string FormatCommandLine(this string value, string? processImage = null) => ProcessNative.FormatCommandLine(value, processImage);
+        
+        /// <summary>
+        ///     Escape commandline arguments so they may be used safely.
+        /// </summary>
+        /// <remarks>
+        ///     See
+        ///     https://blogs.msdn.microsoft.com/twistylittlepassagesallalike/2011/04/23/everyone-quotes-command-line-arguments-the-wrong-way/
+        /// </remarks>
+        /// <param name="args">The arguments to concatenate.</param>
+        /// <returns>The escaped arguments, concatenated.</returns>
+        internal static string EscapeArguments(this string[] args) => string.Join(" ", args.Select(EscapeSingleArg));
```

Comment:
Is this only used for display / ease of copying the command string into a terminal? I want to make sure we're not doing the same thing as in Bebop.

Reply:
This is visual only.